pkg/models/constants.go

```diff
@@ -101,3 +101,23 @@ const (
 	CreditSponsor              CreditType = "SPONSOR"
 	CreditOther                CreditType = "OTHER"
 )
+
+func KnownEcosystems() []Ecosystem {
```

Comment:
I don't think we want to remove it from pkg/lockfile now as it may have some dependency issues for other repos.
But yeah, I added new comments to indicate that both functions return the same set. Thanks!

Reply:
This should be kept within `lockfile` as it's a reflection of what ecosystems it supports "just extracting from" - we only actually use it ourselves for testing to ensure that all the "just work" extractors are accounted for.

Technically it would be a breaking change to remove but also frankly I don't think it hurts having it around because 1. it very much belongs to `lockfiles`, 2. the `semantic` tests use it so it can't be made a test helper within `lockfile` (and outside would break 1.), and 3. because we don't use it in any runtime code it doesn't actually bloat the cli.

(still we'll revisit it generally as part of v2 either way)

Reply:
Thanks @G-Rath!  Yeah, keeping everything within the `lockfile` is ideal. But I encountered a problem when trying to use `internal/semantic` for https://github.com/google/osv-scanner/pull/704. A cyclic import issue showed up (`semantic` imported `lockfile` for ecosystems related). To fix this, I switched to importing `Ecosystem` from `pkg/model`, bypassing the `lockfile` importing for now. if we don't duplicate `KnownEcosystems()` here, the `semantic` tests will fail. Do you have any other ideas to bypass this circular import issue for now.

Reply:
You just need to cast the ecosystem in the `semantic` tests - see https://github.com/google/osv-scanner/pull/713

Reply:
Ah, I didn't notice that tests are in `semantic_test` package, not `semantic`. Thanks!